app/packages/registration-schema/page-manager.ts

```diff
@@ -0,0 +1,42 @@
+import Changeset from 'ember-changeset';
+import { getSchemaBlockGroup, SchemaBlock, SchemaBlockGroup } from 'ember-osf-web/packages/registration-schema';
+
+type ResponseValue = string | string[] | FileReference[] | null;
+
+interface FileReference {
+    id: string;
+    name: string;
+}
+
+export class PageManager {
+    changeSet?: Changeset;
+    schemaBlockGroups?: SchemaBlockGroup[];
+    pageValidations?: any;
+    pageResponses?: Record<string, ResponseValue>;
+    pageIsValid?: boolean;
+    pageHeadingText?: string;
+
+    constructor(pageSchemaBlocks: SchemaBlock[]) {
+        this.schemaBlockGroups = [];
+        for (const block of pageSchemaBlocks) {
+            if (block.schemaBlockGroupKey) {
+                if (!this.schemaBlockGroups.find(group => group.schemaBlockGroupKey === block.schemaBlockGroupKey)) {
+                    this.schemaBlockGroups.push(getSchemaBlockGroup(pageSchemaBlocks, block.schemaBlockGroupKey));
+                }
+            } else {
+                this.schemaBlockGroups.push({
+                    labelBlock: block,
+                });
+            }
+        }
```

Comment:
This can be logic can be replaced with `getSchemaBlockGroups(pageSchemaBlocks)` now